### PR TITLE
Cypress Custom command for Login

### LINKED ISSUE #4225: Feat (E2E): Use Login App commands instead of UI Operation
Use API for login instead of UI login to speed up E2E test scenarios

### PR DESCRIPTION
**Pull request by xmano**
Closes #4225

### CODE REVIEW

**Comment by xmano** [comment]
Tests passing locally!

image

File: apps/catalog/catalog-e2e/src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+declare namespace Cypress {
+  interface Chainable<Subject> {
+    login(email: string, password: string): Chainable<any>;
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog-e2e/src/index.d.ts:3]: Maybe specify what the "any". I guess it's always a user or something like that no ?

File: apps/catalog/catalog-e2e/src/integration/dashboard/import.spec.ts [hunk 2]
```diff
@@ -21,18 +19,18 @@ const MOVIE_IMPORT_TIMEOUT = 120 * SEC;
...
+  cy.login(users[0].email, users[0].password);
+  cy.visit('/c/o/marketplace/home');
 
   cy.location('pathname', {timeout: 120 * SEC})
     .should('include', '/marketplace/home');
 
   cy.log("Switch to dashboard home");
   //Reach dashboard home
-  cy.get('catalog-marketplace button[test-id=menu]', {timeout: 3 * SEC})
+  cy.get('catalog-marketplace button[test-id=menu]', {timeout: 60 * SEC})
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog-e2e/src/integration/dashboard/import.spec.ts:31]: 60sec is a long time no ? Why do we need to wait that long ?

File: apps/catalog/catalog-e2e/src/support/commands.ts [hunk 3]
```diff
@@ -25,3 +28,22 @@ import 'cypress-mailosaur';
 //
 // -- This will overwrite an existing command --
 // Cypress.Commands.overwrite("visit", (originalFn, url, options) => { ... })
+
+//Init the app
+const app = firebase.initializeApp(firebaseConfig('catalog'));
+
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog-e2e/src/support/commands.ts:34]: you'll need to connect to the emulator here I think if you want to login with the emulator : 
```typescript
const auth = app.auh();
auth.useEmulator("http://localhost:9099");
```

Comment (GrandSchtroumpf) [reply apps/catalog/catalog-e2e/src/support/commands.ts:34]: maybe you'll need to have a check on whether or not to use the emulator. Maybe @george43g can answer that.

Comment (xmano) [reply apps/catalog/catalog-e2e/src/support/commands.ts:34]: I will do a more thorough E2E test with the Emulator later. The scope of this PR is to get the custom command syntax, code structure correct.

We know that the tests are working with the login service command. Next what remains is to make sure that it works with Emulator as well and add any additional code required to handle that.

**Review by GrandSchtroumpf — APPROVED** [review]

File: apps/catalog/catalog-e2e/src/index.d.ts (hunk 1 above)
Comment (xmano) [reply apps/catalog/catalog-e2e/src/index.d.ts:3]: import firebase from 'firebase/app';

declare namespace Cypress {
  interface Chainable<Subject> {
    login(email: string, password: string): Chainable<firebase.auth.UserCredential>;
    logout(): Chainable<void>;
  }
}

**Pull request merged**